### ISSUE TITLE
Restriction to Pytorch <= 2.3 on LUMI due to degradation of training

### ISSUE THREAD

**Issue by dietervdb-meteo** [ecmwf/anemoi-core#560]
When upgrading from torch2.3.0-rocm6.0.3 to torch2.7.1-rocm6.2.4 an unexplained change in loss evolution during training appears. This was previously observed in torch-rocm versions in between the above as well. It can be illustrated in very simple models where it leads to a small but significant increase in training loss and change of training trajectory. For larger models it can lead to divergence rather than convergence of training.

See https://github.com/ecmwf/anemoi-core/issues/523#issuecomment-3298733716 for a concrete example.

This difference does not exist on cuda (NVIDIA) devices, e.g. atos (ecmwf HPC) and leonardo. The torch 2.3.0 version on lumi reproduces the loss evolution observed accross torch versions on atos/leonardo.

For this reason only torch <=2.3.0 can can currently be considered consistent on lumi.

**Comment by dietervdb-meteo** [comment]
The precise issue has now been identified #1041 and a solution proposed #1042 .
